src/screens/ENSIntroSheet.tsx

```diff
@@ -39,40 +49,115 @@ export default function ENSIntroSheet() {
+  const { accountENS } = useAccountProfile();
+
+  const { ownedDomains, primaryDomain, nonPrimaryDomains } = useMemo(() => {
+    const ownedDomains = domains?.filter(
+      ({ owner }) => owner.id?.toLowerCase() === accountAddress.toLowerCase()
+    );
+    return {
+      nonPrimaryDomains:
+        ownedDomains?.filter(({ name }) => accountENS !== name) || [],
+      ownedDomains,
+      primaryDomain: ownedDomains?.find(({ name }) => accountENS === name),
+    };
+  }, [accountAddress, accountENS, domains]);
+
+  const uniqueDomain = useMemo(() => {
+    return primaryDomain
+      ? primaryDomain
+      : nonPrimaryDomains?.length === 1
+      ? nonPrimaryDomains?.[0]
+      : null;
+  }, [nonPrimaryDomains, primaryDomain]);
 
-  const ownedDomains = useMemo(
-    () =>
-      domains?.filter(
-        ({ owner }) => owner.id?.toLowerCase() === accountAddress.toLowerCase()
-      ),
-    [accountAddress, domains]
-  );
   const { navigate } = useNavigation();
+  const { startRegistration } = useENSRegistration();
+
   const handleNavigateToSearch = useCallback(() => {
+    startRegistration('', REGISTRATION_MODES.CREATE);
     navigate(Routes.ENS_SEARCH_SHEET);
-  }, [navigate]);
+  }, [navigate, startRegistration]);
 
-  const { startRegistration } = useENSRegistration();
-
-  const handleSelectExistingName = useCallback(() => {
-    const navigateToAssignRecords = (ensName: string) => {
+  const navigateToAssignRecords = useCallback(
+    (ensName: string) => {
       params?.onSelectExistingName?.();
       startRegistration(ensName, REGISTRATION_MODES.EDIT);
       setTimeout(() => {
         navigate(Routes.ENS_ASSIGN_RECORDS_SHEET);
       }, 0);
-    };
+    },
+    [navigate, params, startRegistration]
+  );
+
+  const handleSelectUniqueDomain = useCallback(() => {
+    !!uniqueDomain && navigateToAssignRecords(uniqueDomain?.name);
+  }, [navigateToAssignRecords, uniqueDomain]);
 
-    if (ownedDomains?.length === 1) {
-      navigateToAssignRecords(ownedDomains[0].name);
-    } else {
-      navigate(Routes.SELECT_ENS_SHEET, {
-        onSelectENS: (ensName: string) => {
-          navigateToAssignRecords(ensName);
+  const handleSelectExistingName = useCallback(() => {
+    navigate(Routes.SELECT_ENS_SHEET, {
+      onSelectENS: (ensName: string) => {
+        navigateToAssignRecords(ensName);
+      },
+    });
+  }, [navigate, navigateToAssignRecords]);
+
+  const menuConfig = useMemo(() => {
+    return {
+      menuItems: [
+        {
+          actionKey: AnotherENSEnum.my_ens,
+          actionTitle: lang.t('profiles.intro.my_ens_names'),
+          icon: {
+            iconType: 'SYSTEM',
+            iconValue: 'rectangle.stack.person.crop',
+          },
+        },
+        {
+          actionKey: AnotherENSEnum.search,
+          actionTitle: lang.t('profiles.intro.search_new_ens'),
+          icon: {
+            iconType: 'SYSTEM',
+            iconValue: 'magnifyingglass.circle.fill',
```

Comment:
Maybe we should use `magnifyingglass.circle`? I think it's more of a convention to use the non-filled ones in context menus?

Reply:
done!